### DISCUSSION TITLE
Snapshot and Order Book

### DISCUSSION THREAD

**Discussion by rdabc** [nkaz001/hftbacktest#114] [General]
Hi,

I have an order book without a snapshot. 
Is a snapshot necessary for backtesting? 
How is a snapshot calculated, and why is it crucial for the following day's analysis?

Thank you,
Regards

**Comment by nkaz001** [comment]
If you only have an incremental market depth feed without the initial snapshot, you will have incomplete market depth for the first few periods. You can rely on a natural refresh, expecting the market depth to eventually be filled by the incremental updates. However, this is not guaranteed, and you may have incomplete market depth. Most crypto exchanges have some flaws in building a complete market depth. Over time, it can reach a certain level of completeness. In contrast, for traditional financial exchanges, such as those using the L3 Market-By-Order feed, the initial snapshot and completeness are far more important.

**Comment by rdabc** [comment]
If I want to back-test over a long period (> 20 days), can I provide one data with no snapshot?
Or I have to provide the data day by day?

**Reply by nkaz001** [reply]
If you have enough memory, it's fine to provide just one data. However, it might be better to use the first few days to create the initial snapshot and start backtesting from that point. This approach can help minimize the impact of incomplete market depth at the beginning of the backtest.
